v2/apiserver/internal/core/logs.go

```diff
@@ -168,6 +170,24 @@ func (l *logsService) Stream(
 				}
 			}
 		}
+
+		// Check to see if we need to look up logs via a specific event ID,
+		// as job may be cached and carried over on a retry event
+		if job.Status != nil && job.Status.LogsEventID != "" {
+			event, err = l.eventsStore.Get(ctx, job.Status.LogsEventID)
+			if err != nil {
+				if _, ok := err.(*meta.ErrNotFound); ok {
+					return nil,
+						fmt.Errorf("error retrieving logs for job %q", job.Name)
+				}
```

Comment:
This is a tricky situation. If the user is looking for the logs of a particular job from a particular event Y that is a retry of event X, but that job in question was inherited from X and X is now gone, and we were to respond by saying 404 "couldn't find X" it's confusing to the user who wasn't actually _looking_ for X. You're now suppressing the confusing message. 👍 on that.

The down side now, however, is that the new error will surface to the end user as a 500 with no real detail.

Raising a `*meta.ErrConflict` with a detailed message might be our best bet here, for lack of a better option.

Also, I'm ok with this happening in a follow-up.

Reply:
Opened a follow-up, so we can consider this resolved for now.

https://github.com/brigadecore/brigade/issues/1403

Reply:
Good call, thanks for creating the follow-up.